### PR TITLE
[GarbageCollector] Add garbage collector into kube-controller-manager

### PR DESCRIPTION
**Pull request by caesarxuchao**
It's disabled by default.

Design doc is at https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/garbage-collection.md

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/26341)

### CODE REVIEW

**Comment by caesarxuchao** [comment]
I think we need this piece for 1.3. @lavalamp

File: cmd/kube-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -153,5 +154,6 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Float32Var(&s.KubeAPIQPS, "kube-api-qps", s.KubeAPIQPS, "QPS to use while talking with kubernetes apiserver")
 	fs.Int32Var(&s.KubeAPIBurst, "kube-api-burst", s.KubeAPIBurst, "Burst to use while talking with kubernetes apiserver")
 	fs.DurationVar(&s.ControllerStartInterval.Duration, "controller-start-interval", s.ControllerStartInterval.Duration, "Interval between starting controller managers.")
+	fs.BoolVar(&s.EnableGarbageCollector, "enable-garbage-collector", s.EnableGarbageCollector, "enables the generic garbage collector. MUST be synced with the corresponding flag of the kube-apiserver. WARNING: the generic garbage collector is an alpha feature")
```
Comment (lavalamp) [inline cmd/kube-controller-manager/app/options/options.go]: s/enables/Enables/

end with a period.

**Comment by lavalamp** [comment]
LGTM one nit.

**Comment by caesarxuchao** [comment]
Address the nit. Applying the lgtm per comment.

**Comment by caesarxuchao** [comment]
Added back the lgtm after trivial rebase.

**Pull request merged**